File: board/src/app/components/card/models/card.ts
export interface Card {
  id: string;
  title: string;
  description: string;
  position: number;
  createdAt: Date;
  updatedAt: Date;
  columnId: string;
}

export interface CardResponse {
  id: string;
  title: string;
  description: string;
  position: number;
  created_at: Date;
  updated_at: Date;
  column_id: string;
}

export interface AddCardRequest {
  title: string;
  description?: string;
  columnId: string;
}

export interface EditCardRequest {
  id: string;
  title: string;
  description: string;
  position: number;
  columnId: string;
}

export interface MoveCardRequest {
  previousCardId?: string;
  nextCardId?: string;
  targetColumnId: string;
}

File: angry/src/main/java/br/com/angryapps/angry/models/Card.java
package br.com.angryapps.angry.models;

import jakarta.persistence.*;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "cards")
public class Card {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID id;
    private String title;
    private String description;
    private double position;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @ManyToOne
    @JoinColumn(name = "column_id")
    private Column column;

    public Column getColumn() {return column;}

    public void setColumn(Column column) {this.column = column;}

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public double getPosition() {
        return position;
    }

    public void setPosition(double position) {
        this.position = position;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public LocalDateTime getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(LocalDateTime updatedAt) {
        this.updatedAt = updatedAt;
    }
}
